includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -1919,6 +1929,51 @@ protected function get_account_branding_icon( $default_value = '' ): string {
 		return $default_value;
 	}
 
+	/**
+	 * Checks if the attached payment intent was successful for the current order.
+	 *
+	 * @param  WC_Order $order Current order to check.
+	 *
+	 * @return array|void A successful response in case the attached intent was successful, null if none.
+	 */
+	protected function check_payment_intent_attached_to_order_succeeded( WC_Order $order ) {
+		$intent_id = (string) $order->get_meta( '_intent_id', true );
+		if ( empty( $intent_id ) ) {
+			return;
+		}
+
+		// We only care about payment intent.
+		$is_payment_intent = 'pi_' === substr( $intent_id, 0, 3 );
+		if ( ! $is_payment_intent ) {
+			return;
+		}
+
+		$intent        = $this->payments_api_client->get_intent( $intent_id );
+		$intent_status = $intent->get_status();
```

Comment:
An exception might occur during the `$api_client->get_intent()` call or if the response was erroneous, while trying to get the intent status. Right now if either happened, the whole payment process would fail, even though the exception would be caught, and the handling will be graceful.

Nevertheless, I'm curious if it makes sense to handle the exception here by just logging it, and returning `null`.